docs/networking.md

```diff
@@ -0,0 +1,175 @@
+# How to use networking
+
+cloud-hypervisor can emulate one or more virtual network interfaces, represented at the hypervisor host by [tap devices](https://www.kernel.org/doc/Documentation/networking/tuntap.txt"). This guide briefly describes how to setup and use networking with cloud-hypevisor.
+
+## Start cloud-hypervisor with net devices
+
+Use one or more `--net` command-line argument from cloud-hypervisor to specify the emulation of one or more virtual NIC's. The example below instructs cloud-hypervisor to emulate for instance 2 virtual NIC's:
+
+```bash
+./cloud-hypervisor \
+    --cpus 4 \
+    --memory "size=512M" \
+    --disk my-root-disk.img \
+    --kernel my-vmlinux.bin \
+    --cmdline "console=ttyS0 reboot=k panic=1 nomodules root=/dev/vda3" \
+    --net tap=ich0,mac=a4:a1:c2:00:00:01,ip=192.168.4.2,mask=255.255.255.0 \
+    --net tap=ich1,mac=a4:a1:c2:00:00:02,ip=10.0.1.2,mask=255.255.255.0
```

Comment:
You can only pass one `--net` to the command line, so this should be:
```
--net tap=ich0,mac=a4:a1:c2:00:00:01,ip=192.168.4.2,mask=255.255.255.0 tap=ich1,mac=a4:a1:c2:00:00:02,ip=10.0.1.2,mask=255.255.255.0
```

Reply:
Hmm... As reported earlier on slack there's by my opinion a small inconsistency between the command line handling of the net command and the capabilities of the device manager code which seems to handle multiple  interfaces (I actually tested it). I was hoping that someone could fix this before the next release since I can't at the moment. In src/main.rs line 108 .multiple(true) is missing. I can maybe also open an issue for this in case you agree.

Reply:
We *do* support multiple interfaces, and the command I highlighted:

```
--net tap=ich0,mac=a4:a1:c2:00:00:01,ip=192.168.4.2,mask=255.255.255.0 \
         tap=ich1,mac=a4:a1:c2:00:00:02,ip=10.0.1.2,mask=255.255.255.0
```

does expose 2 interfaces to the guest.

Passing 2 `--net` errors out.

Reply:
Ok.! That explains the confusion. I'll change it. I still believe that you could argue that this syntax can be a bit confusing. qemu for instance also uses -disk A -disk B -disk C etc in case you have more than 1. here with cloud-hypervisor we have similar situations with device and net for instance. keeping the existing approach is fine but then perhaps it should be called -nets and -devices instead to make it more intuitive. But: I'll rest my case, :)